homeassistant/components/barry/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Adds config flow for Barry integration."""
+# pylint: disable=attribute-defined-outside-init
+from pybarry import Barry, InvalidToken
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+
+class BarryConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Barry integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        data_schema = vol.Schema({vol.Required(CONF_ACCESS_TOKEN): str})
+
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        if user_input is not None:
+            access_token = user_input[CONF_ACCESS_TOKEN].replace(" ", "")
+
+            barry_connection = Barry(
+                access_token=access_token,
+            )
+
+            errors = {}
+            try:
+                barry_connection.get_all_metering_points(check_token=True)
```

Comment:
Is this call not blocking io? Should this be async or using the hass executor? See https://developers.home-assistant.io/docs/asyncio_working_with_async/#calling-sync-functions-from-async.

Reply:
Currently, we are not using async calls in the initial release and planning to use them in the future release.

Reply:
Well, the point is, this is an async method, doing a sync call, thus needs to be run in the executor.